app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PostsController < ApplicationController
+
+  def index
+    @posts = Post.all
+  end
+
+  def show
+    redirect_to Post.find(params[:id]).link
```

Comment:
This means that if I navigate to `/posts/:id` I'll get the submitted article instead? That seems counter intuitive to me. I'd expect to see the comment section for that post.

Reply:
Good point, I need to rework that

Reply:
I don't think you need a redirect at all. I think in your `_post` partial that renders the front page items, you'll want something along the lines of `link_to @post.link`

Reply:
I phrased that poorly, I have it fixed like how you described it ^